saleor/graphql/core/mutations.py

```diff
@@ -102,12 +102,22 @@ def __init_subclass_with_meta__(
         cls._update_mutation_arguments_and_fields(
             arguments=arguments, fields=fields)
 
+    @classmethod
+    def _delete_instance(cls, instance):
```

Comment:
I like this idea, we were thinking something similar in "create" and "update" mutations, so that you don't have to override the whole `mutate` method which contains a lot of boilerplate. I think we could rename it to `delete_instance` - since we want to give the ability to override it, it shouldn't be "private" (according to Python's naming conventions).